code/modules/mining/machine_processing.dm

```diff
@@ -172,9 +172,9 @@
 			return
 
 		id.forceMove(get_turf(src))
-		if(!usr.get_active_hand()) //Nothing in the user's hand.
-			usr.put_in_hands(id)
-
+		if(usr.Adjacent(src)
+			if(!usr.get_active_hand()) //Nothing in the user's hand.
```

Comment:
I'm almost certain this check is already in `put_in_hands()` and therefore redundant